### ISSUE TITLE
[Feature Request] Generation from .sol contract.

### ISSUE THREAD

**Issue by galkin** [krzkaczor/TypeChain#61]
Hi, @krzkaczor.

**Do you have plan add generation from `.sol` file?**

Right now I using such flow:

1. Generate `abi` from `sol` file.
2. Generate `typechain files' from `abi`.

There is simplified cli script what I use for step 1. 

```javascript
#!/usr/bin/env node

const { execSync } = require('child_process');
const { readFileSync, writeFileSync } = require('fs');
const { compile } = require('solc');

//Looks like cli arguments
const contractFilePath = 'sol/file/path/should/be/here/';
const contractName = 'ContractNameFromSolFile';
const abiFilePath = 'abi/file/path/should/be/here/';

const contractFile = readFileSync(contractFilePath);
const solcInstance = compile(contractFile.toString());
writeFileSync(abiFilePath, solcInstance.contracts[contractName].interface);
console.log('ABI is compiled');
```

**Comment by krzkaczor** [comment]
I think this is out of the scope of TypeChain project (it would require us to add solc dep etc) but it would be a great fit as standalone cli package. 

Idea: Please publish it as a separate package like idk: `typechain-from-sol` (or some smarter name :D) and I will link it in the readme.

**Comment by krzkaczor** [comment]
@galkin probably it would be even nicer if TypeChain would expose simple API... (no need to run, `execSync`) Let me know what do you think.

**Comment by galkin** [comment]
> Idea: Please publish it as a separate package like idk: typechain-from-sol (or some smarter name :D) and I will link it in the readme.

Deal!

**Comment by galkin** [comment]
> probably it would be even nicer if TypeChain would expose simple API... (no need to run, execSync) Let me know what do you think.

It will be useful. Could you provide this?

**Comment by krzkaczor** [comment]
@galkin sure. I am gonna do this today,  (gimme ~2h). The way I imagine it to work is that you will be able to write something like this:

```typescript
import { generateTypechainWrappers } from "typechain";

async function main() {
  await generateTypeChainWrapper({
    glob: "...".
    // all other CLI options goes here
  });
  // it returns Promise<void>
}
```

it would not return you generated contracts (at least not for now) since typechain needs to copy runtime etc but generate wrappers and do everything as CLI would do. I would expect that in future we could make API more powerful.

What do you think?

**Comment by galkin** [comment]
Let's try!

**Comment by krzkaczor** [comment]
It's done and released as `0.2.4`. There is a readme section added how to use TypeChain as API.

**Comment by krzkaczor** [comment]
I am gonna close it due to inactivity.
